src/components/Header.jsx

```diff
@@ -16,7 +16,7 @@ const Header = ({ setContent ,isVisible,handleVisible}) => {
         handleVisible={handleVisible}
         setShowId={setShowId}
       />
-      <SelectEpisodes isVisible={isVisible} showId={showId} />
+      {isVisible && <SelectEpisodes showId={showId} />}
```

Comment:
My first change is to get rid of `isVisible` as a prop to be passed to the component.

The reason for doing this is, if you don't want to display the `SelectEpisodes` component, you are going through some code in `src/components/SelectEpisodes.jsx` and creating a state and doing a fetch and ... suddenly getting to a ternary check on the return which says `return null` because `isVisible` is `false`.

With my suggestion you don't go through all that fuss!